pkg/ccl/backupccl/restore_job.go

```diff
@@ -692,6 +694,32 @@ func backedUpDescriptorWithInProgressImportInto(
 	return true, nil
 }
 
+// validateRestoredDescriptors returns an error if any descriptor in the cluster is invalid.
+func validateRestoredDescriptors(ctx context.Context, execConfig *sql.ExecutorConfig) error {
+	allDescs, err := backupresolver.LoadAllDescs(ctx, execConfig, hlc.Timestamp{WallTime: timeutil.Now().
+		UnixNano()})
+	if err != nil {
+		return err
+	}
+	targetDescs := make([]catalog.Descriptor, 0)
+	for _, desc := range allDescs {
+		if desc.Dropped() {
+			// For some reason, ValidateDescriptors fails on the temp system table
+			// when they're in the dropped stage.
+			continue
+		}
+	}
+	ok, invalidMsg, err := backuputils.ValidateDescriptors(ctx, targetDescs,
+		execConfig.Settings.Version.BinaryVersion())
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return errors.Newf("Failed descriptor validation check: %s", invalidMsg)
```

Comment:
[nit] Here and elsewhere, I think our preferred error style is to start with a lowercase letter.  I might go with the slightly terser: "descriptor validation failed: %s"